File: 1829/Solution.java
class Solution {
    public int[] getMaximumXor(int[] nums, int maximumBit) {
        int x = 0, n = nums.length, max = (1 << maximumBit) - 1;

        for(int i : nums) x ^= i;

        int[] a = new int[n];
        for(int i = 0; i < n; i++){
            a[i] = x ^ max;
            x ^= nums[n-i-1];
        }
        return a;
    }
}

File: 1829/Solution.rs
impl Solution {
    pub fn get_maximum_xor(nums: Vec<i32>, maximum_bit: i32) -> Vec<i32> {
        let mut x = 0;
        let n = nums.len();
        let max = (1 << maximum_bit) -1;
        for i in 0..n {
            x ^= nums[i];
        }
        let mut a = vec![0; n];
        
        for i in 0..n {
            a[i] = x ^ max;
            x = x ^ nums[n - i - 1];
        }
        a
    }
}
